src/app/state/actions.ts

```diff
@@ -270,3 +297,236 @@ export function removeStoryAsset(assetName: string) {
+    toast.error(errorMessage);
+    return [];
+  }
+}
+
+export function loadAllMyStoriesData(isAuthenticated: boolean) {
+  const store = getDefaultStore();
+
+  // Set loading state
+  store.set(MyStoriesRequestStateAtom, { status: 'loading' });
+
+  // Fetch all data in parallel
+  Promise.all([
+    fetchMyStoriesData('session', false, isAuthenticated),
+    fetchMyStoriesData('session', true, isAuthenticated),
+    fetchMyStoriesData('state', false, isAuthenticated),
+    fetchMyStoriesData('state', true, isAuthenticated),
+  ])
+    .then(([sessionsPrivate, sessionsPublic, statesPrivate, statesPublic]) => {
+      store.set(MyStoriesDataAtom, {
+        'sessions-private': sessionsPrivate,
+        'sessions-public': sessionsPublic,
+        'states-private': statesPrivate,
+        'states-public': statesPublic,
+      });
+      store.set(MyStoriesRequestStateAtom, { status: 'success' });
+    })
+    .catch((error) => {
+      console.error('Error loading my stories data:', error);
+      store.set(MyStoriesRequestStateAtom, { status: 'error', error: error.message });
+    });
+}
+
+export async function loadSession(sessionId: string) {
+  const store = getDefaultStore();
+  try {
+    store.set(IsSessionLoadingAtom, true);
+
+    const response = await authenticatedFetch(`${API_CONFIG.baseUrl}/api/session/${sessionId}/data`);
+
+    if (!response.ok) {
+      throw new Error(`Failed to fetch session data: ${response.statusText}`);
+    }
+
+    const sessionResponse = await response.json();
+    const storyData = sessionResponse;
+
+    if (storyData?.story) {
+      store.set(StoryAtom, storyData.story);
+      store.set(CurrentViewAtom, { type: 'story-options', subview: 'story-metadata' });
+    } else {
+      throw new Error('No story data found in session');
+    }
+  } catch (err) {
+    console.error('Error loading session:', err);
+    const errorMessage = err instanceof Error ? err.message : 'Failed to load session';
+    toast.error(errorMessage);
+  } finally {
+    store.set(IsSessionLoadingAtom, false);
+  }
+}
+
+export async function openItemInBuilder(router: ReturnType<typeof useRouter>, item: Session | State) {
+  try {
+    // For sessions, try to load story data into the builder
+    if (item.type === 'session') {
+      router.push(`/builder/?sessionId=${item.id}`);
+    } else if (item.type === 'state') {
+      // For states, open in external MVS Stories viewer (states are MVS data, not story format)
+      const url = `https://molstar.org/demos/mvs-stories/?story-url=${API_CONFIG.baseUrl}/api/${item.type}/${item.id}/data?format=mvsj`;
+      window.open(url, '_blank');
+    } else {
+      toast.error('Unknown item type');
+    }
+  } catch (err) {
+    console.error('Error opening item:', err);
+    toast.error(err instanceof Error ? err.message : 'Failed to open item');
+  }
+}
+
+// Delete Actions
+export async function deleteSession(sessionId: string, isAuthenticated: boolean) {
```

Comment:
There should be an `updateSession` action as well to be able to overwrite existing sessions.

Similarly, should add a way for the users to modify existing shared stories? I can imagine users will make mistakes and will want to correct them without having to change a link to an existing story.